environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentLoadBalancerService.java

```diff
@@ -67,6 +86,32 @@ public void updateLoadBalancerInEnvironmentAndStacks(EnvironmentDto environmentD
         reactorFlowManager
             .triggerLoadBalancerUpdateFlow(environmentDto, environment.getId(), environment.getName(), environment.getResourceCrn(),
                 environmentLbDto.getEndpointAccessGateway(), environmentLbDto.getEndpointGatewaySubnetIds(), userCrn);
+
+        FlowLogResponse flowLogResponse = getLatestLoadBalancerUpdateFlowConfigLogs(environment.getResourceCrn());
+        if (flowLogResponse == null || hasFlowFailed(List.of(flowLogResponse))) {
+            String errorMsg;
+            if (flowLogResponse == null) {
+                errorMsg = String.format("Unable to find %s flow for tracking", LoadBalancerUpdateFlowConfig.class.getSimpleName());
+            } else {
+                response.setFlowId(flowLogResponse.getFlowId());
+                errorMsg = String.format("Flow in %s state. Please review environment logs.", flowLogResponse.getCurrentState());
+            }
+            LOGGER.error(errorMsg);
+            response.setError(errorMsg);
+            response.setStatus(LoadBalancerUpdateStatus.FAILED);
```

Comment:
Because the environment service's `ReactorFlowManager` does not return a flow id when you initiate a flow. So I don't know a way to get the flow id outside of fetching the logs for the flow, and pulling it from there. If there's some other way to get the flow id in the environment service I missed, please let me know.

Since I needed to pull the flow logs to get the flow id, I figured I might as well check the flow state since I have the logs. This flow only has two steps. The first makes some database updates are is generally quite quick, and the second kicks off some flows in the stack service that do take a long time. Realistically I don't think there's much of a chance that the first step of the flow will have either succeeded or failed when we do this check, so if there's another way to get the flow id I can remove this section. But as long as I have the logs I feel like it's worth checking them before I send the response back.

Reply:
then make it return. you can check this commit where it was introduced for datalake service and fms: https://github.com/hortonworks/cloudbreak/commit/ab5e6b6e0a2716b032c90ceb8631c1c77201c06d
I think this would suit your usecase better and it would be a reusable feature for upcoming changes

Reply:
Thank you! That was what I wanted to do originally, but I wasn't sure how. That example was exactly what I needed :).

I've made some updates to the `EventSender` class in the latest push. I made it return `Optional<FlowIdentifier>`. It looked like there was a case where it wouldn't find a `FlowAcceptResult` in the event data and continued normally, so I didn't want to make it throw an error in that case in case that would cause problems elsewhere in the code. Let me know if you think those changes make sense.